packages/html-reporter/src/headerView.tsx

```diff
@@ -49,12 +49,12 @@ export const HeaderView: React.FC<React.PropsWithChildren<{
       <form className='subnav-search' onSubmit={
         event => {
           event.preventDefault();
-          navigate(`#?q=${filterText ? encodeURIComponent(filterText) : ''}`);
+          navigate(filterText ? '#?' + new URLSearchParams({ q: filterText }) : '/');
```

Comment:
Why `/`? Wouldn't that navigate to the top-level, even when I open/serve the html report from a subpath?

Reply:
That's a fair point. I tried `''`, but that didn't do anything. Let me try that again.

Reply:
there you go: https://github.com/microsoft/playwright/pull/33754/commits/ebf15622e4cf2fb2473543343a7935ad0198a899